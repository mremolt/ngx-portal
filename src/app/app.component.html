<div class="container">
  <div class="header">
    <h1>Welcome to {{ title }}!</h1>
  </div>

  <div class="main"><router-outlet></router-outlet></div>

  <div class="sidebar"><mr-sidebar></mr-sidebar></div>

  <div class="footer"><div>The Footer</div></div>
</div>
