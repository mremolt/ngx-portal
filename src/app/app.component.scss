.container {
  box-sizing: border-box;
  height: 100vh;
  width: 100vw;

  display: grid;
  grid-template-columns: 1fr 1fr 1fr minmax(250px, 1fr);
  grid-template-rows: 100px 1fr auto 100px;
  grid-template-areas:
    'header header header header'
    'main main main main'
    'sidebar sidebar sidebar sidebar'
    'footer footer footer footer';
  grid-gap: 1em;

  div {
    padding: 1em;
  }

  .header {
    grid-area: header;
  }

  .main {
    grid-area: main;
  }

  .sidebar {
    grid-area: sidebar;
    padding: 0;
  }

  .footer {
    grid-area: footer;
    display: flex;
    align-items: center;
    justify-content: center;
    font-size: 1.5em;
  }

  > div {
    border: 1px solid #ccc;
    border-radius: 0.2em;
    background-color: #eee;
  }
}

@media (min-width: 768px) {
  .container {
    grid-template-rows: 100px 1fr 100px;
    grid-template-areas:
      'header header header header'
      'main main main sidebar'
      'footer footer footer footer';
  }
}

@media (min-width: 1200px) {
  .container {
    width: 1200px;
    margin: auto;
  }
}
