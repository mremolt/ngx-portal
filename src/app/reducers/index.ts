import { RouterAction, RouterReducerState, routerReducer } from '@ngrx/router-store';
import { Action, ActionReducer, ActionReducerMap, MetaReducer } from '@ngrx/store';
import { storeLogger } from 'ngrx-store-logger';

import { environment } from '../../environments/environment';

export interface RootState {
  router: RouterReducerState;
}

export interface AnyAction extends Action {
  payload?: any;
  meta?: object;
}

export function logger(reducer: ActionReducer<RootState>): any {
  return storeLogger({ collapsed: true })(reducer);
}

export const reducers: ActionReducerMap<RootState, RouterAction<any>> = {
  router: routerReducer,
};

export const metaReducers: MetaReducer<RootState>[] = !environment.production ? [logger] : [];
