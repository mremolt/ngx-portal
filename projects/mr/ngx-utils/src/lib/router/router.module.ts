import { NgModule } from '@angular/core';
import { EffectsModule } from '@ngrx/effects';
import { RouterEffects } from './router.effects';

@NgModule({
  imports: [EffectsModule.forFeature([RouterEffects])],
})
export class RouterModule {}
