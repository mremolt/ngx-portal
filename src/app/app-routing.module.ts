import { NgModule } from '@angular/core';
import { RouterModule, Routes } from '@angular/router';

import { AuthGuard } from './auth/reducers/auth.guard';

const routes: Routes = [
  {
    path: 'dashboard',
    loadChildren: './dashboard/dashboard.module#DashboardModule',
    canActivateChild: [AuthGuard],
  },
];

@NgModule({
  imports: [RouterModule.forRoot(routes)],
  exports: [RouterModule],
})
export class AppRoutingModule {}
