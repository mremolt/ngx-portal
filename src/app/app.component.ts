import { Component, OnInit } from '@angular/core';

@Component({
  selector: 'mr-root',
  templateUrl: './app.component.html',
  styleUrls: ['./app.component.scss'],
})
export class AppComponent implements OnInit {
  title = 'NGX-Portal';

  constructor() {}

  public ngOnInit() {
    console.log(`Site load: ${performance.now().toFixed(2)} ms!`);
  }
}
