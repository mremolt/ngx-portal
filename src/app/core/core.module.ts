import { CommonModule } from '@angular/common';
import { HttpClientModule } from '@angular/common/http';
import { NgModule } from '@angular/core';
import { EffectsModule } from '@ngrx/effects';

import { environment } from '../../environments/environment';
import { AppEffects } from '../app.effects';
import { AuthModule } from '../auth/auth.module';
import { ReducersModule } from '../reducers/reducers.module';
import { APP_ENVIRONMENT } from '../tokens';

@NgModule({
  imports: [
    // Angular
    CommonModule,
    HttpClientModule,
    // ngrx
    ReducersModule,
    EffectsModule.forRoot([AppEffects]),
    // Features
    AuthModule,
  ],
  declarations: [],
  providers: [
    {
      provide: APP_ENVIRONMENT,
      useValue: environment,
    },
  ],
})
export class CoreModule {}
