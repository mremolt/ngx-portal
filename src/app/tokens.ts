import { InjectionToken } from '@angular/core';

import { Environment } from '../environments/environment';

export const APP_ENVIRONMENT = new InjectionToken<Environment>('APP_ENVIRONMENT');
