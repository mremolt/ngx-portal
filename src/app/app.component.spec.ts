import { TestBed, ComponentFixture } from '@angular/core/testing';
import { RouterTestingModule } from '@angular/router/testing';
import { NO_ERRORS_SCHEMA } from '@angular/core';

import { AppComponent } from './app.component';
import { configureTestBedSuite } from '../testing';

describe('AppComponent', () => {
  configureTestBedSuite();

  let fixture: ComponentFixture<AppComponent>;

  beforeAll(async () => {
    TestBed.configureTestingModule({
      imports: [RouterTestingModule],
      declarations: [AppComponent],
      schemas: [NO_ERRORS_SCHEMA],
    }).compileComponents();
  });

  beforeEach(() => {
    fixture = TestBed.createComponent(AppComponent);
    fixture.detectChanges();
  });

  it('should create the app', () => {
    fixture = TestBed.createComponent(AppComponent);
    const app = fixture.debugElement.componentInstance;
    expect(app).toBeTruthy();
  });

  it(`should have as title 'ngx-portal'`, () => {
    const app = fixture.debugElement.componentInstance;
    expect(app.title).toEqual('NGX-Portal');
  });

  it('should render title in a h1 tag', () => {
    const compiled = fixture.debugElement.nativeElement;
    expect(compiled.querySelector('h1').textContent).toContain('Welcome to NGX-Portal!');
  });

  it('renders the component', () => {
    expect(fixture).toMatchSnapshot();
  });
});
