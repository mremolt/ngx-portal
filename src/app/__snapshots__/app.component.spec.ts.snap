// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppComponent renders the component 1`] = `
<mr-root
  title={[Function String]}
>
  <div
    class="container"
  >
    <div
      class="header"
    >
      <h1>
         Welcome to NGX-Portal! 
      </h1>
    </div>
    <div
      class="main"
    >
       Content 
      <router-outlet />
    </div>
    <div
      class="sidebar"
    >
      <mr-sidebar />
    </div>
    <div
      class="footer"
    >
      <div>
        The Footer
      </div>
    </div>
  </div>
</mr-root>
`;
